pkg/relay/relay.go

```diff
@@ -209,48 +212,82 @@ func (rs *Relay) GetHeader(ctx context.Context, m *structs.MetricGroup, request
 		return nil, ErrNoBuilderBid
 	}
 
-	bid := types.BuilderBid{
-		Header: header.Header,
-		Value:  header.Trace.Value,
-		Pubkey: rs.config.PubKey,
-	}
+	fork := rs.beaconState.GetFork(uint64(slot.Epoch()))
 
-	tSignature := time.Now()
-	signature, err := types.SignMessage(&bid, rs.config.BuilderSigningDomain, rs.config.SecretKey)
-	m.AppendSince(tSignature, "getHeader", "signature")
-	if err != nil {
-		return nil, ErrInternal
-	}
+	if fork == structs.ForkBellatrix {
```

Comment:
Why not switch cases?

Reply:
Because it's too big to be fair, We can pack this into functions potentially, and do switch but this after that PR